data_preprocessing/name_preprocessing.py

```diff
@@ -0,0 +1,230 @@
+     based on organization associated with document
+
+     Adds people to db_current if their alias is associated with a doc but is not in
+     db_current yet
+
+     Ignores:
+         organizations not in RAW_ORG_TO_CLEAN_ORG_DICT
+         docs with more than 1 organization
+         docs with more than 4 recipients
+         docs with no people associated with them
+
+
+    :param csv_path: Path
+    :param db_to_add: PeopleDatabase
+    :return:
+    """
+
+    if not people_db:
+        people_db = PeopleDatabase()
+
+    alias_to_person_dict = people_db.get_alias_to_person_dict()
+
+    # get lists of dicts for authors and recipients. each of them has 3 fields:
+    # general, person, organization
+    people_dicts = get_au_and_rc_by_document(csv_path, return_type='both')
+
+    for person in people_dicts:
+
+        orgs = set()
+        for org in person['organization']:
+            if org not in RAW_ORG_TO_CLEAN_ORG_DICT:
+                continue
+            orgs.add(RAW_ORG_TO_CLEAN_ORG_DICT[org])
+
+        if len(orgs) != 1:
+            continue
+        org = list(orgs)[0]
+        aliases = person['person']
+        aliases.extend(person['general'])
+        if not aliases or len(aliases) >= 4:
+            continue
+
+        for alias in aliases:
+            if alias not in alias_to_person_dict:
+                people_db.add_person_raw(alias)
+            else:
+                alias_to_person_dict[alias].positions[org] += 1
+
+
+def load_documents_to_dataframe(path) -> pd.DataFrame:
+    """
+    Loads the Dunn documents into a pandas Dataframe
+    :return: pd.DataFrame
+    """
+    df = pd.read_csv(path).fillna('')  # pylint: disable=C0103
+    return df
+
+
+def get_au_and_rc_by_document(path, return_type='both') -> list:
+    """
+    Creates a list of documents such that each element consists of a dict with keys
+
+    returns either just information about the authors, the recipients, or both
+
+    Note(SR): to give authors and recipients the same structure, I have renamed 'au' and 'rc'
+    to general.
+
+    'au', 'au_org', 'au_person' OR 'rc', 'rc_org', 'rc_person'.
+    These keys map to info about the authors/recipients and organizations associated with docs
+    :param path: Path
+    :param return_type: str
+    :return: list
+    """
+
+    if return_type not in ['authors', 'recipients', 'both']:
+        raise ValueError(f'get_au_and_rc_by_document can only be called with return_type "both",'
+                         f'"authors," or "recipients".')
+
+    df = load_documents_to_dataframe(path)  # pylint: disable=C0103
+
+    authors_by_docs = []
+    for _, row in df.iterrows():
+        authors_by_docs.append({
+            'general': parse_column_person(row['au']),
+            'organization': parse_column_org(row['au_org']),
+            'person': parse_column_person(row['au_person'])
+        })
+    recipients_by_docs = []
+    for _, row in df.iterrows():
```

Comment:
No need to iterate over the df twice. 
Just move recipients_by_docs to the first loop